src/main.rs

```diff
@@ -518,17 +555,22 @@ impl AsyncComponent for App {
         model.menu_popover = widgets.menu_popover.clone();
 
         if init.minimize_on_start {
-            root.minimize();
+            root.hide();
         }
 
+        root.connect_close_request(|view| {
+            view.hide();
+            gtk::glib::Propagation::Stop
+        });
```

Comment:
I think this is more cleanly done with [`set_hide_on_close`](https://docs.gtk.org/gtk4/method.Window.set_hide_on_close.html), but this should be un-done if we fail to initialize tray icon as described in other comments.